### PR TITLE
added eventer for deployment event status updates

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/1616.

### CODE REVIEW

File: flag/service/eventer/eventer.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package eventer
+
+import (
+	"github.com/giantswarm/draughtsman-operator/flag/service/eventer/github"
+)
+
+type Eventer struct {
```
Comment (xh3b4sd) [inline flag/service/eventer/eventer.go]: This is all the same as in the eventer or draughtsman itself. Exception is we only need the eventer type, the Github oauth token and the Github organisation.

File: main.go [hunk 2]
```diff
@@ -136,19 +138,6 @@ func mainWithError() error {
...
-	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.TLS.CrtFile, "", "Certificate file path to use to authenticate with Kubernetes.")
-	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.TLS.KeyFile, "", "Key file path to use to authenticate with Kubernetes.")
-
-	daemonCommand.PersistentFlags().String(f.Service.Installer.Helm.HelmBinaryPath, "/bin/helm", "Path to Helm binary. Needs CNR registry plugin installed.")
-	daemonCommand.PersistentFlags().String(f.Service.Installer.Helm.Organisation, "", "Organisation of Helm CNR registry.")
-	daemonCommand.PersistentFlags().String(f.Service.Installer.Helm.Password, "", "Password for Helm CNR registry.")
-	daemonCommand.PersistentFlags().String(f.Service.Installer.Helm.Registry, "quay.io", "URL for Helm CNR registry.")
-	daemonCommand.PersistentFlags().String(f.Service.Installer.Helm.Username, "", "Username for Helm CNR registry.")
-	daemonCommand.PersistentFlags().String(f.Service.Installer.Type, string(helm.HelmInstallerType), "Which installer to use for installation management.")
-
```
Comment (xh3b4sd) [inline main.go]: Boyscouting the order here.

File: service/resource/project/resource.go [hunk 3]
```diff
@@ -190,6 +198,10 @@ func (r *Resource) ProcessCreateState(obj, createState interface{}) error {
 		for _, p := range projectsToCreate {
 			instErr := r.installer.Install(installerspec.Project{Name: p.Name, Ref: p.Ref})
 			if instErr != nil {
+				evenErr := r.eventer.SetFailedStatus(eventerspec.DeploymentEvent{ID: 0, Name: p.Name, Sha: p.Ref})
```
Comment (xh3b4sd) [inline service/resource/project/resource.go]: The ID has to be properly set. I need to do this in the next step. For this we need to add the deployment event ID to the TPO and wire it through the stack.

File: main.go (hunk 2 above)
Comment (JosephSalisbury) [reply main.go]: It's depressing how bad i was at the ordering of these :D

**Pull request merged**